src/FtlServer.cpp

```diff
@@ -154,7 +173,28 @@ void FtlServer::onNewControlConnection(std::unique_ptr<ConnectionTransport> conn
         std::move(ingestControlConnection));
     ingestControlConnectionPtr->StartAsync();
 
-    spdlog::info("New FTL control connection is pending.");
+    spdlog::info("New FTL control connection is pending from {}", addrString);
+
+    // If this connection doesn't successfully auth in a certain amount of time, close it.
+    auto timeoutThread = std::thread([this, ingestControlConnectionPtr, addrString]() {
+        std::unique_lock threadLock(threadShutdownMutex);
+        threadShutdownConditionVariable.wait_for(threadLock,
+            std::chrono::milliseconds(CONNECTION_AUTH_TIMEOUT_MS));
+        if (isStopping)
```

Comment:
I think there are race conditions here with `isStopping`, maybe it would be simpler/better to move access to be protected by the mutex?

It's an extreme edge case, but I think this thread could read the atomic as `false` and get context switched out for the server thread which is executes `FtlServer->Stop()`, sets the atomic to true takes the lock on the `streamDataMutex` and clears all the pending connections.

The end effect is the same with the current logic, the only worry would be if future changes run into this race.